modified:   src/automerge.py

F-WRunTime opened the pull request:
- Simplified Check: The updated code retrieves the combined status of the latest commit directly.
- Single Condition: It checks if the combined status state is 'success' to determine if all checks have passed.
- Logging: It logs a single message if the checks are not passing, indicating the overall state.

### Benefits of the Change:

- Efficiency: The new approach reduces the complexity by avoiding the need to manually iterate over each required check.
- Readability: The code is more concise and easier to understand, focusing on the overall status rather than individual checks.
- Performance: By using the combined status, it potentially reduces the number of API calls and processing required.
- This change streamlines the process of determining if a PR is ready to be merged based on its status checks, making the code more maintainable and efficient.


Minimum requirements 
![image](https://github.com/user-attachments/assets/a04e60f5-1b88-43e2-81f5-1c1a1bbdb886)
![image](https://github.com/user-attachments/assets/0ea722b2-85d7-45cf-9cd9-a078deb495cf)
![image](https://github.com/user-attachments/assets/937c48d8-d3f4-41ab-99ff-72bfd8e19589)
![image](https://github.com/user-attachments/assets/c07fbcbf-e874-4dc5-9b77-87764eab1552)

F-WRunTime committed:
modified:   src/automerge.py
- Modify api end points used to check status. Previous did not have a
  permission option with fine-grained tokens.
- This reduces permissioni requirements to use automerger

F-WRunTime committed:
modified:   .github/workflows/test.yaml
- Update tests to use the new less permissive token.

F-WRunTime committed:
modified:   src/automerge.py
- Checking the 'chech' statuses per commit
- Ccombined commit was not
  always accurate to the mergeable state of  a PR due to annotations and
other items checked not important to basic check passed? Approved? Up to
datae? Yes? Merge.

tothtamas28 commented on src/automerge.py:
```diff
@@ -90,21 +90,13 @@ def run_git_command(command_args: str) -> subprocess.CompletedProcess:
-                    all_checks_passed = False
+        commit = pr.get_commits().reversed[0]
+        statuses = commit.get_statuses()
+        all_checks_passed = all(print(status) for status in statuses)
```
This evaluates to `False` regardless of `statuses`. (And as a side effect, will print the first `status`.)

FWBlueWizard replied:
The tests came back as expected. 
https://github.com/runtimeverification/automerge/actions/runs/12061856900/job/33634749102#step:4:1

I'll debug this now. Must be getting a false positive

FWBlueWizard replied:
Addressed this issue. This required reworking permissions to introduce the use for a Github App. 
Remaining is to update the README and docs and tests

F-WRunTime committed:
modified:   src/automerge.py
- Rework logic.
- New dependency for infrastructure to run. Install Github App
  AutomergePR Permission Control to have access to check statuses. Bug
in Github fine-grained tokens or deprecated documentaiton never removed.

F-WRunTime committed:
modified:   .github/workflows/test.yaml
- Updating workflow test to use the GH APP credentials
modified:   README.md
- Modifying instructions and informating need for github and what
  permissions are needed for the github App

F-WRunTime committed:
Rearrange Github app token step and use app token for check steps

F-WRunTime committed:
Missed first job adding 'token'

F-WRunTime committed:
References to token were mixed up between jobs and steps outside originating job

F-WRunTime committed:
Generate token for all repositories under current owner running the action

F-WRunTime committed:
Generate a new token for each job

tothtamas28 commented on README.md:
````diff
@@ -112,8 +129,9 @@ Checkout the repository you wish to run automerge on to a local directory.
 ```
-
-Now you need to run the command from this new directory 
+Setup GITHUB_TOKEN with the appropriate permissions. Content Read/Write, Pull Request Read/Write, adminstration Read, Checks Read. 
````
```suggestion
Setup `GITHUB_TOKEN` with the appropriate permissions: Content Read/Write, Pull Request Read/Write, Adminstration Read, Checks Read. 
```

F-WRunTime committed:
Update README.md

Grammar fix in readme.

Co-authored-by: Tamás Tóth <tothtamas28@users.noreply.github.com>
